src/lib/autoinstall/autosetup_helpers.rb

```diff
@@ -193,8 +193,37 @@ def autosetup_country
       Yast::Profile.remove_sections("language") if Yast::Profile.current.key?("language")
     end
 
+    # Invokes autoyast setup for firewall
+    def autosetup_firewall
+      return if !Yast::Profile.current["firewall"]
+      return if need_second_stage_run?
+
+      log.info("Importing Firewall settings from AY profile")
+      Yast::WFM.CallFunction("firewall_auto", ["Import", Yast::Profile.current["firewall"]])
+
+      Yast::Profile.remove_sections("firewall")
+    end
+
   private
 
+    # Checks whether we need to run second stage handling
+    def need_second_stage_run?
+      Yast.import "Linuxrc"
+
+      profile = Yast::Profile.current
+
+      # We have a problem when
+      # 1) running remote installation
+      # 2) second stage was requested
+      # 3) firewall was configured (somehow) and started via AY profile we can expect that
+      # ssh / vnc port can be blocked.
```

Comment:
I would add a **TODO** here as we wanted to configure it always in the first stage but at the end decided to not do in these cases.